app/controllers/api/alert_definitions_controller.rb

```diff
@@ -1,26 +1,39 @@
 module Api
   class AlertDefinitionsController < BaseController
-    REQUIRED_FIELDS = %w(description db expression options).freeze
+    REQUIRED_FIELDS = %w(description db options).freeze
 
     before_action :set_additional_attributes
 
     def create_resource(type, id, data = {})
       assert_id_not_specified(data, type)
       assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
+      if expression_values(data).size != 1
```

Comment:
This looks like business logic to me. What's more - it's duplicated on line 26. This should be added to the model as a validation, which will greatly simplify this code. The code here should be almost entirely concerned with deserializing the JSON request body and passing data to the model.

Reply:
removed that, will add a validation in the model later on.